test/hotspot/jtreg/runtime/cds/appcds/ClassListWithInvokerClassNoSource.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+
+public class ClassListWithInvokerClassNoSource {
```

Comment:
I think we should add comments like this to describe the purpose of this class:
```
// When this class is executed, the following classes will be generated by
// java.lang.invoke.ClassSpecializer with the code source
// "_ClassSpecializer_generateConcreteSpeciesCode".
//
//   - java.lang.invoke.BoundMethodHandle$Species_LFDIIL
//   - java.lang.invoke.BoundMethodHandle$Species_LFDIILJ
//   - java.lang.invoke.BoundMethodHandle$Species_LFDIILJD
//   - ...
// 
// The TestDumpClassListSource test case checks that these classes are not
// written into the classlist, as they cannot be handled by -Xshare:dump
```

Also, the name `ClassListWithInvokerClassNoSource` is misleading, since these classes DO have a code source, except the code source cannot be understood by -Xshare:dump.

Maybe we should rename it to something like `ClassSpecializerTestApp`, to describe what this class actually does?